RecoHGCal/TICL/plugins/TrackstersPCA.cc

```diff
@@ -52,10 +56,21 @@ void ticl::assignPCAtoTracksters(std::vector<Trackster> &tracksters,
       fillPoint(layerClusters[trackster.vertices(i)], weight);
       for (size_t j = 0; j < 3; ++j)
         barycenter[j] += point[j];
+
+      // Also compute timing
+      float timeE = layerClustersTime.get(trackster.vertices(i)).second;
+      if (timeE > -1.) {
+	times.push_back(layerClustersTime.get(trackster.vertices(i)).first);
+	timeErrors.push_back(1. / pow(timeE, 2));
+      }
     }
```

Comment:
I'm not sure the algebra here is correct, especially for all the cases in which a single layer cluster is used more than once. This would be especially true for all the track-seeded hadronic trackster for which the flag to compress them into a single trackster is True.
In all those cases, if I'm not wrong, the layer clusters used more than once are correctly weighted by the PCA and barycenter calculation, but will enter the timing with their full information over and over again, depending on their usage.
I'm not totally sure how to handle these cases if extending the weighted average is something that would make sense or not.

Reply:
Good point. I have to think...
Because on one side, the time of the hits and the resolution on them depend on the total energy deposited, even if this comes through overlapping showers. 
But indeed, in case a layer cluster belongs mostly to trackster n.1 for example, it should be counted mainly in the average time of trackster n.1 and very little in the others.
So, probably extending the same logic of the weights is ok, in particular if the weights are non democratic (1/total times of sharing)

Reply:
I think the situation here is even more complicated... and mainly due to the way in which we've implemented the counting, merging and weights.
When we merge 2 tracksters, we simply append the entries of one trackster to the entries of the other. This has the nice property of conserving the energy in absolute terms, w/o the need to loop over all other tracksters that could potentially share some layer clusters with any of the 2 being merged. 
The drawback of this is the complexity of keeping track of the real usage of a layer cluster. If a layer cluster is used by both when the 2 are merged it will be counted twice. The situation can only get worse if the multiplicity is greater than 2.
I think this has to be critically reviewed and better understood, also in contexts that have nothing to do with timing.

Reply:
Ah yes. Then I suggest to live things as they were, with timing for trackster from standard seeding, and re thing about the merging and timing for merged later.